packages/nx/src/command-line/migrate.ts

```diff
@@ -393,15 +393,14 @@ function parseTargetPackageAndVersion(args: string) {
       return { targetPackage, targetVersion };
     }
   } else {
-    if (
-      args.match(/^\d+(?:\.\d+)?(?:\.\d+)?$/) ||
-      args === 'latest' ||
-      args === 'next'
-    ) {
+    // adapted from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+    // to support partial version numbers
+    const semverRegex =
+      /^(?:0|[1-9]\d*)(?:\.(?:0|[1-9]\d*))?(?:\.(?:0|[1-9]\d*))?(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/;
+    if (args.match(semverRegex) || args === 'latest' || args === 'next') {
```

Comment:
Huh... I wonder if we should lean more on `semver.valid` and fill in the partials?

```suggestion
    if (
      // Valid semver version
      valid(args) ||
      // latest or next
      args === 'latest' || args === 'next' ||
      // Partial versions: (ex. 1, 1.2)
      args.match(/^\d+(?:\.\d+)?(?:\.\d+)?$/)) {
```

Reply:
Totally, way simpler.